### PR TITLE
Step 6 for the server tutorial is completed.

### PR DESCRIPTION
**Pull request by Phoebe-Khokgawe**
The website can handle user adding comments and displaying the number for comments depending on the user's choice.
The delete all comments button is added to delete all the comments in the data store.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -32,22 +40,29 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    ArrayList<String> jsonString = new ArrayList<>();
-    addToList();
-    for(String obj : strings){
-        jsonString.add(convertToJSON(obj));
+
+    Query query = new Query("Comments").addSort("comment", SortDirection.DESCENDING);
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: WIll this sort comments alphabetically?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -32,22 +40,29 @@
...
 
-    response.setContentType("text/html;");
-    int randomIndex = (int)Math.floor(Math.random() * ((int)jsonString.size()));
-    System.out.print(jsonString.size() + " randome num: " + randomIndex);
-    response.getWriter().println(jsonString.get(randomIndex).toString());
+    Gson gson = new Gson();
+
+    response.setContentType("application/json");
+    response.getWriter().write(gson.toJson(toSendOver));
+    response.sendRedirect("/postRequest.html");
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: why redirect, isnt this making it go to another page?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Sorry I've done some touch up after the pull request. In this commit I have updated the comments to be displayed based on the time it's been submitted https://github.com/Phoebe-Khokgawe/STEP_Starter/pull/9/commits/f74d300f381d94330b947b5ff24e9ef63405083f.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comments").addSort("comment", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+        datastore.delete(entity.getKey());
+    }
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:42]: Is there a way to bulk delete? ie only 1 call to datastore

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -36,16 +36,16 @@
...
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     Query query = new Query("Comments").addSort("comment", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
     ArrayList<String> toSendOver = new ArrayList<>();
+    toSendOver.add(String. valueOf(numOfComments));
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I dont think you need to send the length, just send the array list

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -62,6 +62,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
     String text = getParameter(request, "comment", "");
+    String quantity = getParameter(request,"quantity","4");
+    numOfComments = Integer.parseInt(quantity.equals("")?"1":quantity);
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: This is saving the value from a post of a form, but this value will be quickly overwritten by the next post from another user.

File: portfolio/.settings/org.eclipse.core.resources.prefs [hunk 6]
```diff
@@ -0,0 +1,3 @@
+eclipse.preferences.version=1
```
Comment (leonnemets) [inline portfolio/.settings/org.eclipse.core.resources.prefs]: gitignore .settings folder

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -41,11 +41,10 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    Query query = new Query("Comments").addSort("comment", SortDirection.DESCENDING);
+    Query query = new Query("Comments").addSort("time", SortDirection.DESCENDING);
```
Comment (leonnemets) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: nice

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -106,11 +106,11 @@ function loadComments(){
             console.log(maxComments + 'Maximum comments');
         const taskListElement = document.getElementById('comments-list');
     //printout the response
-    //try normal for loop
-        const displayedComments = maxComments < comments.length ? maxComments : comments.length;
+        //number of comments to display
+        const displayedComments = maxComments == 0? comments.length : maxComments < comments.length ? maxComments : comments.length;
```
Comment (leonnemets) [inline portfolio/src/main/webapp/script.js:97]: line too long. Run a code formatter in your IDE

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: This is the old code. It's fix now.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: This line has been removed

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (leonnemets) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: sorry, I was using github reviews incorrectly. My analogies to internal code tooling were wrong. I talked to a colleague and I understand better

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java (hunk 3 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:42]: I've tried doing a lot of researches on this but I couldn't really find a way to bulk delete. Can you point me in the right direction please.

File: portfolio/.settings/org.eclipse.core.resources.prefs (hunk 6 above)
Comment (Phoebe-Khokgawe) [reply portfolio/.settings/org.eclipse.core.resources.prefs]: fixed in 1c5f413fe6c41160d8b05b41b46102e2d9df161b

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (Phoebe-Khokgawe) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Yes I don't need to save the value. numOfComments is now a local variable. 9ecf11367307858c121189ed146b2a1c3ac88af3

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java (hunk 3 above)
Comment (leonnemets) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:42]: Seems like it is not straight forward. This is the kind of thing I was thinking about - https://cloud.google.com/datastore/docs/bulk-delete.

Dont worry about it for this exercise

**Review by leonnemets — APPROVED** [review]
Nice work

**Pull request merged**